packages/optimistic-oracle/index.js

```diff
@@ -48,15 +47,18 @@ async function run({
       optimisticOracleProposerConfig,
     });
 
+    // If bot is not running on mainnet, then assume that the network's "Oracle" in the Finder is a SinkOracle.
+    const isMainnet = Boolean((await web3.eth.getChainId()) === 1);
+
     // Create the OptimisticOracleClient to query on-chain information, GasEstimator to get latest gas prices and an
     // instance of the OO Proposer to respond to price requests and proposals.
     const optimisticOracleClient = new OptimisticOracleClient(
       logger,
       getAbi("OptimisticOracle"),
-      getAbi("Voting"),
+      isMainnet ? getAbi("Voting") : getAbi("SinkOracle"),
```

Comment:
We can probably change this to a mapping once we have other non-Voting Oracles, for rollup networks for example.